app/jobs/add_registration_job.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+class AddRegistrationJob < ApplicationJob
+  def perform(lane_name, competition_id, user_id, lane_params)
+    lane_model_name = lane_name.upcase_first
+
+    lane = Registrations::Lanes::class_eval(lane_model_name)
+    if lane.nil?
+      raise "No Lane exists for #{lane_model_name}"
+    end
+
+    ActiveRecord::Base.transaction do
+      lane.process!(lane_params[:competing], user_id, competition_id)
```

Comment:
Nitpick: You don't know (in general) what kind of lane this will be, so you cannot assume that the key `competing` is relevant here.
You should just pass the whole params to the lane and let it do its own thing.

Reply:
yeah, I even changed this in my API branch.